File: 01/c.py
def isPrime(x):
    if x < 2:
        return False
    elif x == 2:
        return True
    if x % 2 == 0:
        return False
    i = 3
    while i * i <= x:
        if x % i == 0:
            return False
        i += 2
    return True


def main():
    cnt = 0
    n = input()
    for i in xrange(n):
        if isPrime(input()):
            cnt += 1

    print cnt

if __name__ == '__main__':
    main()

File: 01/c.cpp
#include<bits/stdc++.h>
using namespace std;

static const int MAX = 1e7;

bool isprime[MAX + 1];

void eratos() {
    memset(isprime, true, sizeof(isprime));
    isprime[0] = isprime[1] = false;

    for (int i = 2; i * i <= MAX; i++) {
        if (isprime[i]) {
            int j = i + i;
            while (j <= MAX) {
                isprime[j] = false;
                j += i;
            }
        }
    }
}

int main() {
    eratos();
    
    int cnt = 0;
    int n; cin >> n;
    for (int i = 0; i < n; i++) {
        int x; scanf("%d", &x);
        cnt += (isprime[x]) ? 1 : 0;
    }
    
    cout << cnt << endl;
    
    return 0;
}
